lib/modules/dosomething/dosomething_api/resources/campaign_resource.inc

```diff
@@ -359,6 +362,23 @@ function _campaign_resource_reportback($nid, $values) {
     watchdog('dosomething_api', '_campaign_resource_reportback:' . json_encode($values));
   }
 
-  return dosomething_reportback_save($values, $user);
+  $transaction_id = $_SERVER['HTTP_X_REQUEST_ID'];
+
+  if (! $transaction_id) {
```

Comment:
I'm not sure I understand this logic. If you have a request _without_ a transaction ID you send to rogue? If you do have one you don't? 

What happens when like the Mobile app sends over RBs when it hasn't implemented transaction ids yet?